cmd/jujud/agent/controllercharm.go

```diff
@@ -342,5 +418,16 @@ func addControllerApplication(
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	// Expose all endpoints of the controller app
+	mappedExposeParams := map[string]state.ExposedEndpoint{
```

Comment:
Controllers on public have a public address. Doesn't this allow anyone to get to any of these endpoints?
```
provides:
  dashboard:
    interface: juju-dashboard
  website:
    interface: http
  metrics-endpoint:
    interface: prometheus_scrape
```

Reply:
I've run the following on Azure with Juju 3.6.4,
```
$ juju bootstrap azure test-ssh
$ curl https://20.164.128.166:17070/introspection/metrics -k
authentication failed: no credentials provided
```
So the metrics endpoint is already exposed, but guarded by basic auth.
Looking at the [controller charm](https://github.com/juju/juju-controller/blob/3.6/src/charm.py), the other 2 endpoints don't seem to go to a specific path beyond the root and don't use any port beyond the API server's port.